Pspcl/Pspcl.Web/Controllers/PreviewController.cs

```diff
@@ -1,14 +1,125 @@
-﻿using Microsoft.AspNetCore.Authorization;
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Pspcl.Core.Domain;
+using Pspcl.Services.Interfaces;
+using Pspcl.Web.ViewModels;
 
 namespace Pspcl.Web.Controllers
 {
-	[Authorize]
-	public class PreviewController : Controller
-	{
-		public IActionResult Preview()
-		{
-			return View();
-		}
-	}
-}
+    [Authorize]
+    public class PreviewController : Controller
+    {
+        private readonly IStockService _stockService;
+        private readonly IMapper _mapper;
+        public PreviewController(IStockService stockService, IMapper mapper)
+        {
+            _stockService = stockService;
+            _mapper = mapper;
+        }
+        [HttpGet]
+        public IActionResult Preview()
+        {
+            var jsonResponse = TempData["StockViewModel"] as string;
+            var stockViewModel = JsonConvert.DeserializeObject<StockViewModel>(jsonResponse);
+            var materialGroupName = _stockService.GetMaterialGroupById(stockViewModel.MaterialGroupId);
+            var materialTypeName = _stockService.GetMaterialTypeById(stockViewModel.MaterialTypeId);
+            var materialCodeName = _stockService.GetMaterialCodeById(stockViewModel.MaterialIdByCode);
+            var ratingName = _stockService.GetRatingNameById(stockViewModel.MaterialTypeId);
+            stockViewModel.SelectedMaterialGroupName = materialGroupName;
+            stockViewModel.SelectedMaterialTypeName = materialTypeName;
+            stockViewModel.SelectedMaterialCodeName = materialCodeName;
+            stockViewModel.SelectedRatingName = ratingName;
+            return View(stockViewModel);
+        }
+        [HttpPost]
+        public IActionResult Preview(IFormCollection formCollection,string save)
+        {
+            try
+            {
+                var model = new StockViewModel();
+                DateTime date = DateTime.Parse(formCollection["GRNDate"]);
```

Comment:
If GRNDate gets null, it will throw exception and will break the page.

Reply:
form collection variable in any means null aya to ?

Reply:
In that case it will go to the catch block .

Reply:
I will put formcollection in if condition